db/migrate/20230621003955_add_names_to_user.rb

```diff
@@ -0,0 +1,6 @@
+class AddNamesToUser < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :first_name, :string, default: 'first_name', null: false
```

Comment:
So this is a good start! There are a couple things to note here. This is one potential way to solve this problem but it comes with the downfalls of allowing a default (which we don't want) and locking the db when we run this (which isn't a big issue for our current scale but something to be aware of).

Ima help out a bit here (I realized a bit late that this is a somewhat harder story than I expected). 

What you are gonna want to do locally first is run `rake db:rollback` to revert these changes.

Then update this file to look like this:
```
  def up
    add_column :users, :first_name, :string, null: true
    add_column :users, :last_name, :string, null: true

    User.update_all(first_name: 'first_name', last_name: 'last_name')

    change_column_null :users, :first_name, false
    change_column_null :users, :last_name, false
  end

  def down
    remove_column :users, :first_name, :string
    remove_column :users, :last_name, :string
  end
  ```
  
  What this is doing is:
  1. we add the new columns to the table, but they are nullable
  2. we backfill all the columns to have some "default" data
  3. we make the columns not null because we have filled in all the data
  
  Sidenote: we do this as up/down because we added some additional stuff to be done.

Reply:
Thank you for this feedback. Can you tell me exactly what locks the db? I suspect it isn't the adding of the columns, but the back fill via the default value?  I will complete these changes, and it's been really cool to learn more about these differences (and why they are really important for future projects/ uses).

Reply:
yes, I am pretty sure the `default:` does bad stuff and should NEVER be used as a way to backfill. (maybe things have changed in recent versions, but this is just my general advice).

Reply:
Ahh. Thank you for this information.

Reply:
This migration won't work for an important reason @toyhammered 

Devise invitable creates the new user on *invite*, not after the user clicks the link in their email and fills out the form. This means you run straight into a not-null PG error when trying to invite a user just by their email.

To have a not-null constraint with no default, I think we have to allow the admin sending the invite to fill in the user's first and last name while filling out the form to invite them. If that's not desirable, I think we have to make it with a default and just let the user change that when they're signing up.